boards/posix/nrf52_bsim/main.c

```diff
@@ -80,6 +82,8 @@ int main(int argc, char *argv[])
 
 	args = nrfbsim_argsparse(argc, argv);
 	global_device_nbr = args->global_device_nbr;
+	s_id = args->s_id;
+	device_nbr = args->device_nbr;
```

Comment:
Please use the `global_device_nbr` instead (This is a bit irrelevant for this device as it only connects to one Phy, but in general they could be different)